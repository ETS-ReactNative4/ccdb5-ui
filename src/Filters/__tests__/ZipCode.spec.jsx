import { shallow } from 'enzyme';
import React from 'react'
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import ReduxZipCode, { mapDispatchToProps, ZipCode } from '../ZipCode'

const fixture = [
  { key: "200XX", doc_count: 9999 },
  { key: "300XX", doc_count: 999 },
  { key: "400XX", doc_count: 99 },
  { key: "500XX", doc_count: 9 }
]

function setupEnzyme() {
  const props = {
    forTypeahead: [],
    options: [],
    typeaheadSelect: jest.fn()
  }

  const target = shallow(<ZipCode {...props} />);

  return {
    props,
    target
  }
}

function setupSnapshot() {
  const middlewares = [thunk]
  const mockStore = configureMockStore(middlewares)
  const store = mockStore({
    query: {
      zip_code: ['300XX']
    },
    aggs: {
      zip_code: fixture
    }
  })

  return renderer.create(
    <Provider store={store}>
      <IntlProvider locale="en">
        <ReduxZipCode />
      </IntlProvider>
    </Provider>
  )
}

describe('component::ZipCode', () => {
  describe('snapshots', () => {
    it('renders without crashing', () => {
      const target = setupSnapshot()
      let tree = target.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  describe('_onOptionSelected', () => {
    it('checks the filter associated with the option', () => {
      const {target, props} = setupEnzyme()
      target.instance()._onOptionSelected({key: 'foo'})
      expect(props.typeaheadSelect).toHaveBeenCalledWith('foo')
    })
  })

  describe('mapDispatchToProps', () => {
    it('hooks into addMultipleFilters', () => {
      const dispatch = jest.fn()
      mapDispatchToProps(dispatch).typeaheadSelect('foo')
      expect(dispatch.mock.calls.length).toEqual(1)
    })
  })
})
