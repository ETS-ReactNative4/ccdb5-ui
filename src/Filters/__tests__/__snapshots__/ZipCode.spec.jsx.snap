exports[`component::ZipCode snapshots renders without crashing 1`] = `
<section
  className="collapsible-filter aggregation">
  <div
    className="layout-row">
    <h5
      className="flex-all">
      Zip Code
    </h5>
    <div
      className="flex-fixed toggle">
      <button
        className="a-btn a-btn__link hover"
        onClick={[Function]}>
        Hide
        <span
          className="cf-icon cf-icon-minus-round" />
      </button>
    </div>
  </div>
  <p>
    The mailing ZIP code provided by the consumer
  </p>
  <section
    className="typeahead "
    onBlur={[Function]}
    onFocus={[Function]}>
    <input
      autoComplete="off"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Enter first three digits of zip code"
      type="text"
      value="" />
  </section>
  <ul>
    <li
      className="flex-fixed layout-row">
      <input
        aria-label="300XX"
        checked={true}
        className="flex-fixed"
        onClick={[Function]}
        type="checkbox" />
      <span
        className="flex-all bucket-key">
        300XX
      </span>
      <span
        className="flex-fixed bucket-count">
        <span>
          999
        </span>
      </span>
    </li>
  </ul>
</section>
`;
